File: Form1.vb
Public Class Calculator
    'muziishere'
    'Global variables to store the current calculated answer and sign'
    Dim answer As Integer
    Dim sign As String

    Private Sub Form1_Load(sender As Object, e As EventArgs) Handles MyBase.Load

    End Sub

    Private Sub btnEqual_Click(sender As Object, e As EventArgs) Handles btnEqual.Click
        Dim nums(2) As Integer
        Dim temp(2) As String

        If sign = "+" Then
            temp = Split(txtDisplay.Text, "+")
            nums(0) = Convert.ToInt32(temp(0))
            nums(1) = Convert.ToInt32(temp(1))
            answer = nums(0) + nums(1)
        ElseIf sign = "-" Then
            temp = Split(txtDisplay.Text, "-")
            nums(0) = Convert.ToInt32(temp(0))
            nums(1) = Convert.ToInt32(temp(1))
            answer = nums(0) - nums(1)
        ElseIf sign = "/" Then
            temp = Split(txtDisplay.Text, "/")
            nums(0) = Convert.ToInt32(temp(0))
            nums(1) = Convert.ToInt32(temp(1))
            answer = nums(0) / nums(1)
        ElseIf sign = "x" Then
            temp = Split(txtDisplay.Text, "x")
            nums(0) = Convert.ToInt32(temp(0))
            nums(1) = Convert.ToInt32(temp(1))
            answer = nums(0) * nums(1)
        End If

        txtDisplay.Text &= Environment.NewLine & Convert.ToString(answer)
    End Sub

    Private Sub btnAdd_Click(sender As Object, e As EventArgs) Handles btnAdd.Click
        sign = "+"
        If answer > 0 Or answer < 0 Then
            txtDisplay.Text = Convert.ToString(answer) & "+"
        Else
            txtDisplay.Text &= "+"
        End If
    End Sub

    Private Sub btnMinus_Click(sender As Object, e As EventArgs) Handles btnMinus.Click
        sign = "-"
        If answer > 0 Or answer < 0 Then
            txtDisplay.Text = Convert.ToString(answer) & "-"
        Else
            txtDisplay.Text &= "-"
        End If
    End Sub
    Private Sub btnDivide_Click(sender As Object, e As EventArgs) Handles btnDivide.Click
        sign = "/"
        If answer > 0 Or answer < 0 Then
            txtDisplay.Text = Convert.ToString(answer) & "/"
        Else
            txtDisplay.Text &= "/"
        End If
    End Sub

    Private Sub btnMultiply_Click(sender As Object, e As EventArgs) Handles btnMultiply.Click
        sign = "x"
        If answer > 0 Or answer < 0 Then
            txtDisplay.Text = Convert.ToString(answer) & "x"
        Else
            txtDisplay.Text &= "x"
        End If
    End Sub

    Private Sub btnRemove_Click(sender As Object, e As EventArgs) Handles btnRemove.Click
        If txtDisplay.Text.Length > 0 Then
            txtDisplay.Text = txtDisplay.Text.Remove(txtDisplay.Text.Length - 1)
        End If
    End Sub

    Private Sub btnClear_Click(sender As Object, e As EventArgs) Handles btnClear.Click
        txtDisplay.Text = ""
        answer = Nothing
        sign = Nothing
    End Sub

    Private Sub btn1_Click(sender As Object, e As EventArgs) Handles btn1.Click
        txtDisplay.Text &= "1"
    End Sub

    Private Sub btn2_Click(sender As Object, e As EventArgs) Handles btn2.Click
        txtDisplay.Text &= "2"
    End Sub

    Private Sub btn3_Click(sender As Object, e As EventArgs) Handles btn3.Click
        txtDisplay.Text &= "3"
    End Sub

    Private Sub btn4_Click(sender As Object, e As EventArgs) Handles btn4.Click
        txtDisplay.Text &= "4"
    End Sub

    Private Sub btn5_Click(sender As Object, e As EventArgs) Handles btn5.Click
        txtDisplay.Text &= "5"
    End Sub

    Private Sub btn6_Click(sender As Object, e As EventArgs) Handles btn6.Click
        txtDisplay.Text &= "6"
    End Sub

    Private Sub btn7_Click(sender As Object, e As EventArgs) Handles btn7.Click
        txtDisplay.Text &= "7"
    End Sub

    Private Sub btn8_Click(sender As Object, e As EventArgs) Handles btn8.Click
        txtDisplay.Text &= "8"
    End Sub

    Private Sub btn9_Click(sender As Object, e As EventArgs) Handles btn9.Click
        txtDisplay.Text &= "9"
    End Sub

    Private Sub btn0_Click(sender As Object, e As EventArgs) Handles btn0.Click
        txtDisplay.Text &= "0"
    End Sub

End Class

File: Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Text.RegularExpressions;
using System.Threading.Tasks;
using System.Windows.Forms;
using org.matheval;

namespace Calculator_CS
{
    public partial class Calculator : Form
    {
        public Calculator()
        {
            InitializeComponent();
        }

        private void Calculator_Load(object sender, EventArgs e)
        {
        }

        private void btnEqual_Click(object sender, EventArgs e)
        {
            Expression expression = new Expression(txtDisplay.Text.Replace("%", "/100").Replace("x", "*"));
            try {
                Object result = expression.Eval();
                txtDisplay.Text = result.ToString();
            }
            catch (DivideByZeroException) {
                txtDisplay.Text = "Cannot divide by zero";
            }
            catch (Exception) {
                Console.WriteLine("Some exception occurred!");
            }
        }
        private void btnAdd_Click(object sender, EventArgs e)
        {
            txtDisplay.Text += "+";
        }

        private void btnMinus_Click(object sender, EventArgs e)
        {
            txtDisplay.Text += "-";
        }
        private void btnDivide_Click(object sender, EventArgs e)
        {
            txtDisplay.Text += "/";
        }

        private void btnMultiply_Click(object sender, EventArgs e)
        {
            txtDisplay.Text += "x";
        }

        private void btnRemove_Click(object sender, EventArgs e)
        {
            if (txtDisplay.Text.Length > 0)
                txtDisplay.Text = txtDisplay.Text.Remove(txtDisplay.Text.Length - 1);
        }

        private void btnClear_Click(object sender, EventArgs e)
        {
            clear();
        }

        private void btn1_Click(object sender, EventArgs e)
        {
            txtDisplay.Text += "1";
        }

        private void btn2_Click(object sender, EventArgs e)
        {
            txtDisplay.Text += "2";
        }
        private void btn3_Click(object sender, EventArgs e)
        {
            txtDisplay.Text += "3";
        }

        private void btn4_Click(object sender, EventArgs e)
        {
            txtDisplay.Text += "4";
        }

        private void btn5_Click(object sender, EventArgs e)
        {
            txtDisplay.Text += "5";
        }

        private void btn6_Click(object sender, EventArgs e)
        {
            txtDisplay.Text += "6";
        }

        private void btn7_Click(object sender, EventArgs e)
        {
            txtDisplay.Text += "7";
        }

        private void btn8_Click(object sender, EventArgs e)
        {
            txtDisplay.Text += "8";
        }

        private void btn9_Click(object sender, EventArgs e)
        {
            txtDisplay.Text += "9";
        }

        private void btn0_Click(object sender, EventArgs e)
        {
            txtDisplay.Text += "0";
        }

        private void darkToolStripMenuItem_Click(object sender, EventArgs e)
        {
            designChange(Color.FromArgb(255, 32, 32, 32),Color.White,Color.Gold);
        }
        private void lightToolStripMenuItem_Click(object sender, EventArgs e)
        {
            designChange(Color.White,Color.Black,Color.Purple);
        }

        // Resets calculator display text
        private void clear()
        {
            txtDisplay.Text = "";
        }

        private void designChange(Color backColor, Color foreColor, Color accentColor)
        {
            this.BackColor = backColor;

            txtDisplay.BackColor = backColor;
            txtDisplay.ForeColor = foreColor;

            foreach (Button btn in this.Controls.OfType<Button>())
            {
                btn.BackColor = backColor;
                btn.ForeColor = foreColor;

                if (!Regex.IsMatch(btn.Text, @"[0-9.%]"))
                {
                    btn.ForeColor = accentColor;
                }
            }
            btnEqual.ForeColor = backColor;
            btnEqual.BackColor = accentColor;
        }
        private void btnDot_Click(object sender, EventArgs e)
        {
            txtDisplay.Text += ".";
        }

        private void btnPercentage_Click(object sender, EventArgs e)
        {
            txtDisplay.Text += "%";
        }

        private void customizeToolStripMenuItem_Click(object sender, EventArgs e)
        {
            Form customize = new Customize();
            if (customize.ShowDialog() == DialogResult.OK)
            {
                designChange(Customize.backColor, Customize.foreColor, Customize.accentColor);
            }
            
        }
    }
}